algo/src/main/java/org/neo4j/graphalgo/ShortestPathDeltaSteppingProc.java

```diff
@@ -70,7 +72,9 @@
                 .load(HeavyGraphFactory.class);
 
         return new ShortestPathDeltaStepping(graph, delta)
-                .withExecutorService(Pools.DEFAULT)
+                .withExecutorService(Executors.newFixedThreadPool(
+                        configuration.getInt("concurrency", 4)
```

Comment:
the concurrency default should come from the config and there it should be inferred from the available processors or global configuration.